### PR TITLE
Increase coverage of arduino related code

### PR DESCRIPTION
**Pull request by jamessynge**
Add ability to shrink size of read chunks and size of read queue,
which will cause some chunks to be dropped, exercising the retry
logic.

Mark some code to be skipped in coverage calculations.

Partially addresses issue #653.

### CODE REVIEW

File: pocs/sensors/arduino_io.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def detect_board_on_port(port, logger=None):
                 return data['name']
             logger.warning('Unable to find board name in reading: {}', reading)
             return None
-        except Exception as e:
+        except Exception as e:  # pragma: no cover
```
Comment (wtgee) [inline pocs/sensors/arduino_io.py]: No way we can trigger it?

File: pocs/serial_handlers/protocol_arduinosimulator.py [hunk 2]
```diff
@@ -30,14 +30,17 @@ class ArduinoSimulator:
...
-                and acted upon. Elements are of type bytes.
-            json_queue: The queue.Queue instance to which json messages (serialized to bytes)
-                are written at ~9600 baud. Elements are of type bytes.
+            relay_queue: The queue.Queue instance from which relay command
+                bytes are read and acted upon. Elements are of type bytes.
+            json_queue: The queue.Queue instance to which json messages
+                (serialized to bytes) are written at ~9600 baud. Elements
+                are of type bytes (i.e. each element is a sequence of bytes of
+                length up to chunk_size).
+            chunk_size: The number of bytes to write to json_queue at a time.
```
Comment (wtgee) [inline pocs/serial_handlers/protocol_arduinosimulator.py]: No default for this?

File: pocs/serial_handlers/protocol_arduinosimulator.py [hunk 3]
```diff
@@ -173,17 +178,15 @@ def __init__(self, *args, **kwargs):
...
+        self.relay_queue = queue.Queue(maxsize=1)
+        self.json_queue = queue.Queue(maxsize=1)
         self.json_bytes = bytearray()
         self.stop = threading.Event()
         self.stop.set()
         self.device_simulator = None
 
     def __del__(self):
-        if self.simulator_thread:
+        if self.simulator_thread:  # pragma: no cover
```
Comment (wtgee) [inline pocs/serial_handlers/protocol_arduinosimulator.py]: This seems like a lot to not cover

File: pocs/sensors/arduino_io.py (hunk 1 above)
Comment (jamessynge) [reply pocs/sensors/arduino_io.py]: No, not that I was able to find.

File: pocs/serial_handlers/protocol_arduinosimulator.py (hunk 2 above)
Comment (jamessynge) [reply pocs/serial_handlers/protocol_arduinosimulator.py]: Yes, but not here, instead down in _create_simulator.

File: pocs/serial_handlers/protocol_arduinosimulator.py (hunk 3 above)
Comment (jamessynge) [reply pocs/serial_handlers/protocol_arduinosimulator.py]: It basically is covering a misuse of the simulator, i.e. of test code.

**Pull request merged**